spectree/spec.py

```diff
@@ -95,7 +97,8 @@ def register(self, app: Any):
         init step.
         """
         self.app = app
-        self.backend.register_route(self.app)
+        if self.publish_apidoc:
+            self.backend.register_route(self.app)
```

Comment:
Maybe we can add this to the README HowTo part? I think this flag is not necessary unless we support more features like "only expose part of the doc". WDYT?

Reply:
It seems to me that other parts rely on the self.app inside spectree so I kept the register part of the code that sets that variable. For example the starlette_plugin.py uses it in find_routes. Tracking down the implications of removing the self.app was too complex for the fix I was trying to do so for safety I decided to keep it. If you are confident that it be the same as omitting the register(app) step then it's fine with me adding it to the README.

Reply:
Thanks for your time. I have double-checked the code, all the `self.app` usages are related to generating the OpenAPI config. Without the `spec.register()`, we should be safe to use the validation feature without binding the OpenAPI endpoints.

Reply:
Is it also the case for the plugins?

Falcon uses self.spectree.app here:
https://github.com/0b01001001/spectree/blob/7d585b6eaef721abc115dfb40851cf9912538c8c/spectree/plugins/falcon_plugin.py#L87-L101

Starlette uses self.spectree.app here:
https://github.com/0b01001001/spectree/blob/7d585b6eaef721abc115dfb40851cf9912538c8c/spectree/plugins/starlette_plugin.py#L143-L179

Are this `find_routes` methods also only for generating the OpenAPI config? I refactored this two plugins because they were using `self.app` on themselves and the `BasePlugin` does not have a self.app. I felt it wasn't very consistent across plugins and the `app` can still be accessed from `self.spectree.app`. This might be a change to keep.

If you give me the confirmation these `find_routes` methods on the plugins are also strictly for OpenAPI generation I'll go ahead and remove the changes and add a new entry to `README.md` documenting that omitting the register(app) will cause the OpenAPI routes to not be generated but the validation will still be enforced.

Also let me know if you want me to keep the `self.app` -> `self.spectree.app` refactor.

P.S.: No need to thank me for my time. It's my pleasure to be able to contribute to the tools I use.

Reply:
- `find_routes` is only used to generate the OpenAPI config.
- Yes, they should not access `self.app` since there is no such attribute.
- You're right. The implementation here is a bit ugly. `SpecTree` and `BasePlugin` can access each other. Need to refactor this part.
- I'm okay with the `self.app` -> `self.spectree.app` refactor. To make it consistent, you can also choose to delete the `app` arg from `register_root()` and use `self.spectree.app` directly.